packages/vite/src/node/optimizer/scan.ts

```diff
@@ -456,3 +502,32 @@ export function shouldExternalizeDep(
   }
   return false
 }
+
+function parseAstro(contents: string): { js: string; html: string } {
```

Comment:
Wanted to declare a function outside of the load callback just for cleanliness, but perhaps this isn’t the best place for this. Can move this anywhere.